docs/README.md

```diff
@@ -0,0 +1,624 @@
+Welcome to the *AIM API - Alpha*!
+
+# Alpha Notice
+
+The AIM API is in an "alpha stage" to gather customer feedback. While in alpha, the API may change in backwards incompatible ways to accommodate for fixes and additions. Breaking changes will be communicated to the primary API contact at least 5 business days in advance.
+
+TSG reserves the right to determine what constitutes a breaking changes. A definition of "breaking changes" will be made available before formal release.
+
+# HTTP API
+
+## API Discovery
+
+In order to reduce client coupling, the AIM API provides an API discovery document available at:
```

Comment:
Al:
> What is client coupling, maybe should have a glossary of certain terms at the end of this document

Reply:
[Coupling](https://en.wikipedia.org/wiki/Coupling_(computer_programming)) describes how dependent one piece of code is on another (or on an API). Essentially, if something is highly coupled to our API, the more likely a change we make would be regarded as a breaking change. That's why we want to reduce coupling - we want to reduce the likelihood that we have to make breaking changes.

In this case, if client code had to hard code every single URL we use, then they'd have a lot of work to do if we ever changed our URL structure (ex: change `/api/warehouse/v1/report` -> `/api/vision/report`). In this case, you could say the client is "highly coupled" to our URL structure - which is what this discovery document aims to reduce. As is, this isn't a huge win, but we can expand the discoverability of our API in the future.